### DISCUSSION TITLE
Can't select a voice program if the USB MIDI keyboard is plugged in

### DISCUSSION THREAD

**Discussion by Colpocorto** [probonopd/MiniDexed#377] [Q&A]
Hi,

I have just built a Minidexed (typical setup: PCM5102, LCD1602, KY-040 encoder, RPI3B) and everything works perfectly except for one thing:  

if I use the encoder to choose a voice program, it won't work if the USB MIDI keyboard is plugged into the RPI3. If I unplug the keyboard, it allows me to select the voice with the encoder. The rest of functionalities work perfectly with the encoder even if the keyboard is plugged.

I guess it is some misconfiguration but can't figure out where the mistake is. According to the documentation it is possible to configure a MIDI controller as a remote control, but I can't find a reason the encoder should stop working if a MIDI keyboard is detected.

Any ideas?

**Thank you very much** in advance.

**Comment by probonopd** [comment]
Hi @Colpocorto, welcome. Sounds like a bug!

Can you attach the Raspberry Pi to a display with HDMI, and take a picture of the output when it works vs. when it doesn't work? Maybe we can figure something out that way.

**Reply by Colpocorto** [reply]
Hi!

It turns out that the HDMI port is broken. I will try to reproduce the issue on another RPI3 but it will be during the weekend (I’m flying out right now).

Thank you for your quick response, by the way!

**Reply by Colpocorto** [reply]
Well, here it is.

I've recorded a video showing the whole issue.

https://youtu.be/wvF3ZHdCWIU

What is shown in the video:

00:00 -> boot the RPI 3 with an encoder and a 16x2 I2C display
00:10 -> browse voices on TG1. No problem at all.
00:30 -> plug a USB MIDI keyboard in
00:40 -> MIDI keyboard is recognized on the HDMI debugging log
00:48 -> the voice selection on TG1 stops working. The system does not respond to the encoder pulses.
00:54 -> Nothing special is logged on the HDMI screen
01:00 -> unplug the USB MIDI keyboard
01:05 -> the voice selection starts working again! Furthermore, all the attempts to twist the encoder seem to be stored in a buffer and now are executed quickly.
01:14 -> nothing special is logged on the HDMI debugging screen (other than the USB MIDI disconnection).

I have reproduced this with two different units (both are an RPI 3B+) and two different PSU. The rest of the hardware (encoder, MIDI keyboard, etc. were the same in both tests).

Any ideas?

**Reply by BobanSpasic** [reply]
1. Test with another USB cable.
2. Post your minidexed.ini file here (if different from the file that you downloaded from here)

**Reply by Colpocorto** [reply]
Hi  @BobanSpasic, thank you for your quick response.

1. Just tried another USB cable. Identical results.
2. Please, find attached my minidexed.ini file

[minidexed.ini.zip](https://github.com/probonopd/MiniDexed/files/9992964/minidexed.ini.zip)

**Reply by Banana71** [reply]
Hi, I would change MIDIButtonCh=17 to MIDIButtonCh=0 first

**Reply by BobanSpasic** [reply]
Just for testing, delete the whole '# MIDI Button Navigation' section from your minidexed.ini and try if your encoder is working without it.

EDIT: or just try what Banana71 wrote

**Reply by Colpocorto** [reply]
Thank you @Banana71 and @BobanSpasic 

I have tried your suggestions. However, the problem persists.

I'm trying a different USB MIDI keyboard and see what happens. Although the keyboard I'm using is a MIDI Class Compliant synthesizer, there could be some kind of incompatibility between both devices.

**Reply by BobanSpasic** [reply]
You could also try some of the previous releases. It can be some regression in the build you are using.

**Reply by Colpocorto** [reply]
Some news: I have tested a different keyboard and it worked as intended. The first keyboard I used has a very simple MIDI implementation (no program change or control change messages at all), that is the only difference (anyway it should work fine too).

Since the encoder issue only happens during the program change operation (it works fine while browsing the other menus), it is likely a regression related to the "menu browsing from the MIDI keyboard" feature.  

I'll try a previous release and report the results of the test here.

Thank you all for your support.

**Comment by BobanSpasic** [comment]
Could you try another power supply?
I have had weird behavior of my RPi3+ with a USB Keyboard, and it was because the power supply was too weak.

**Reply by Colpocorto** [reply]
I will try; in any case, I'm using a 3A power supply and the USB MIDI Keyboard is NOT powered via USB (it has its own PSU).

**Comment by Colpocorto** [comment]
Additional info:

- I have tested and monitored the MIDI keyboard on a regular PC with MIDI OX. All the MIDI events received from the keyboard were absolutely normal. No unexpected events, no missing bytes... everything is perfect.
- Even if the voice selection does not work, the whole synth works perfectly. MIDI events are received by MiniDexed and notes coming from the MIDI keyboard are played flawlessly.

**Comment by Colpocorto** [comment]
Hi,

I have tested some of the previous releases (in fact, all of them except 2022-04-16, that seems to be not compatible with the 16x2 LCD working in I2C mode).

Exactly the same results.
